reference-implementation/lib/transform-stream.js

```diff
@@ -97,40 +105,51 @@ function TransformStreamErrorInternal(transformStream, e) {
   }
 }
 
-function TransformStreamReadyPromise(transformStream) {
-  assert(transformStream._backpressureChangePromise !== undefined);
+// Used for preventing the next write() call on TransformStreamSink until there
+// is no longer backpressure.
+function TransformStreamReadableReadyPromise(transformStream) {
+  assert(transformStream._backpressureChangePromise !== undefined,
+         '_backpressureChangePromise should have been initialized');
 
-  if (transformStream._readableBackpressure === false) {
+  if (transformStream._backpressure === false) {
     return Promise.resolve();
   }
 
-  transformStream._backpressureChangePromise.then(backpressure => assert(backpressure === false));
+  assert(transformStream._backpressure === true, '_backpressure should have been initialized');
 
   return transformStream._backpressureChangePromise;
 }
 
 function TransformStreamSetBackpressure(transformStream, backpressure) {
   // console.log(`TransformStreamSetBackpressure(${backpressure})`);
 
-  assert(transformStream._readableBackpressure !== backpressure);
+  // Passes also when called for initialization.
+  assert(transformStream._backpressure !== backpressure,
+         'TransformStreamSetBackpressure() should be called only when backpressure is changed');
 
   if (transformStream._backpressureChangePromise !== undefined) {
+    // The fulfillment value is just for the assertion.
```

Comment:
"the assertion" is ambiguous because it's not clear which assertion is being referred to. How about "just for a sanity check"?